File: src/main/java/ar/edu/utn/frba/tacs/tp/api/herocardsgame/utils/FileConstructorUtils.java
package ar.edu.utn.frba.tacs.tp.api.herocardsgame.utils;

import com.google.gson.Gson;

import java.io.FileReader;
import java.io.IOException;
import java.io.Reader;

public class FileConstructorUtils {

    public static <T> T createFromFile(String file, Class<T> classReturn) {

        Gson gson = new Gson();

        try (Reader reader = new FileReader(file)) {
            return gson.fromJson(reader, classReturn);
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }
}

File: src/test/kotlin/ar/edu/utn/frba/tacs/tp/api/herocardsgame/utils/FileConstructorUtils.kt
package ar.edu.utn.frba.tacs.tp.api.herocardsgame.utils;

import com.google.gson.Gson
import java.io.File
import java.io.IOException

class FileConstructorUtils {
    companion object {
        fun <T> createFromFile(file: String, classReturn: Class<T>): T {
            try {
                val bufferedReader = File(file).bufferedReader()
                return Gson().fromJson(bufferedReader, classReturn)
            } catch (e: IOException) {
                throw RuntimeException(e);
            }
        }

        fun <T> createFromContent(content: String, classReturn: Class<T>): T {
            try {
                return Gson().fromJson(content, classReturn)
            } catch (e: IOException) {
                throw RuntimeException(e);
            }
        }
    }
}